tensorflow-examples/src/main/java/org/tensorflow/model/examples/mnist/training/LinearRegressionExample.java

```diff
@@ -0,0 +1,154 @@
+            Assign<TFloat32> biasInit = tf.assign(bias, tf.constant(1f));
+
+            // Define the model function weight*x + bias
+            Mul<TFloat32> mul = tf.math.mul(X, weight);
+            Add<TFloat32> yPredicted = tf.math.add(mul, bias);
+
+            // Define loss function MSE
+            Pow<TFloat32> sum = tf.math.pow(tf.math.sub(yPredicted, Y), tf.constant(2f));
+            Div<TFloat32> mse = tf.math.div(sum, tf.constant(2f * N));
+
+            // Define gradient operators
+            List<Variable<TFloat32>> variables = new ArrayList<>();
+            variables.add(weight);
+            variables.add(bias);
+
+            Gradients gradients = tf.gradients(mse, variables);
+
+            Constant<TFloat32> alpha = tf.constant(0.2f);
+
+            ApplyGradientDescent<TFloat32> weightGradientDescent = tf.train.applyGradientDescent(weight, alpha, gradients.dy(0));
+            ApplyGradientDescent<TFloat32> biasGradientDescent = tf.train.applyGradientDescent(bias, alpha, gradients.dy(1));
+
+            try (Session session = new Session(graph)) {
+                // Initialize graph variables
+                session.runner()
+                        .addTarget(weightInit)
+                        .addTarget(biasInit)
+                        .run();
+
+                // Train the model on data
+                for (int i = 0; i < xValues.length; i++) {
+                    float y = yValues[i];
+                    float x = xValues[i];
+
+                    Tensor<TFloat32> xTensor = TFloat32.scalarOf(x);
+                    Tensor<TFloat32> yTensor = TFloat32.scalarOf(y);
+
+                    session.runner()
+                            .addTarget(weightGradientDescent)
+                            .addTarget(biasGradientDescent)
+                            .feed(X.asOutput(), xTensor)
+                            .feed(Y.asOutput(), yTensor)
+                            .run();
+
+                    System.out.println("Training phase");
+                    System.out.println("X is " + x + " Y is " + y);
+                }
+
+                // Extract the weight value
+                Tensor<TFloat32> weightValue = session.runner()
+                        .fetch("Variable")
+                        .run().get(0).expect(TFloat32.DTYPE);
+
+                System.out.println("Weight is " + weightValue.data().getFloat());
+
+                // Extract the bias value
+                Tensor<TFloat32> biasValue = session.runner()
+                        .fetch("Variable_1")
+                        .run().get(0).expect(TFloat32.DTYPE);
+
+                System.out.println("Bias is " + biasValue.data().getFloat());
+
+                // Let's predict y for x = 10f
+                float x = 10f;
+                float predictedY = 0f;
+
+                Tensor<TFloat32> xTensor = TFloat32.scalarOf(x);
+                Tensor<TFloat32> yTensor = TFloat32.scalarOf(NO_MEANING_VALUE_TO_PUT_IN_PLACEHOLDER);
```

Comment:
Why do you need to feed in `yTensor`?

Reply:
Need to double-check (I thought that it doesn't work withouth feeding the yTensor)

Reply:
If you don't request `sum` or `mse` it should work just fine without `yTensor` and if it doesn't there's a bug somewhere.